src/main/scala/com/softwaremill/realworld/users/UsersEndpoints.scala

```diff
@@ -0,0 +1,67 @@
+package com.softwaremill.realworld.users
+
+import com.softwaremill.realworld.db.{Db, DbConfig}
+import com.softwaremill.realworld.utils.*
+import io.getquill.SnakeCase
+import sttp.model.StatusCode
+import sttp.tapir.generic.auto.*
+import sttp.tapir.json.zio.jsonBody
+import sttp.tapir.server.ServerEndpoint.Full
+import sttp.tapir.ztapir.*
+import sttp.tapir.{EndpointInput, PublicEndpoint, Validator}
+import zio.json.{DeriveJsonDecoder, DeriveJsonEncoder}
+import zio.{Cause, Exit, ZIO, ZLayer}
+
+import javax.sql.DataSource
+
+class UsersEndpoints(usersService: UsersService, base: BaseEndpoints):
+
+  import UsersEndpoints.given
+
+  val get: ZServerEndpoint[Any, Any] = base.secureEndpoint.get
+    .in("api" / "user")
+    .out(jsonBody[User])
+    .serverLogic(session =>
+      _ =>
+        usersService.findById(session.id).logError.mapError {
+          case _: Exceptions.NotFound => NotFound()
+          case _                      => InternalServerError()
+        }.map(User.apply)
+    )
+
+  val post: ZServerEndpoint[Any, Any] = base.publicEndpoint.post
+    .in("api" / "users")
+    .in(jsonBody[UserRegister])
+    .out(jsonBody[User])
+    .serverLogic(data =>
+      usersService
+        .registerNewUser(data.user)
+        .logError
+        .mapError {
+          case _: Exceptions.Conflict => Conflict()
+          case _                      => InternalServerError()
+        }
+        .either
```

Comment:
something is wrong here, ".either" should be removed - I assume ZIO (not RIO) should be expected here and this is somehow determined by type of "publicEndpoint"

Reply:
I think once you migrate to `.zServerEndpoint` this will solve itself